tests/test_metadata_harvester_cli.py

```diff
@@ -28,75 +11,268 @@ def kielipankki_api_url():
-               "access_rights": {
-        "license": [
-            {
-                "url": "http://uri.suomi.fi/codelist/fairdata/license/code/undernegotiation"
-            }
-        ],
-        "access_type": {
-            "access_type": "http://uri.suomi.fi/codelist/fairdata/access_type/code/open"
-        }
+    shared_request_mocker.get(kielipankki_api_url, text=single_record_xml)
+    yield {"urn.fi/urn:nbn:fi:lb-2017021609": {"data_catalog": "urn:nbn:fi:att:data-catalog-kielipankki-v4", "language": [{"url": "http://lexvo.org/id/iso639-3/fin"}], "field_of_science": [{"url": "http://www.yso.fi/onto/okm-tieteenala/ta112"}], "persistent_identifier": "urn.fi/urn:nbn:fi:lb-2017021609", "title": {"en": "Silva Kiuru's Time Expressions Corpus", "fi": "Silva Kiurun ajanilmausaineisto"}, "description": {"en": "This corpus of time expressions has been compiled from literary works, translations, dialect texts as well as other texts. Format: word documents.", "fi": "Tämä suomen kielen ajanilmauksia käsittävä aineisto on koottu kaunokirjallisten alkuperäisteosten, käännösten, murreaineistojen ja muiden tekstien pohjalta."}, "modified": "2017-02-15T00:00:00.000000Z", "issued": "2017-02-15T00:00:00.000000Z", "access_rights": {"license": [{"url": "http://uri.suomi.fi/codelist/fairdata/license/code/undernegotiation"}], "access_type": {"url": "http://uri.suomi.fi/codelist/fairdata/access_type/code/open"}}}}
+
+
+def test_records_to_dict_with_last_harvest_date(single_record_to_dict, create_test_log_file):
+    """
+    Test that fetching records based on a date in log file succeeds (only updated records are
+    fetched).
+    """
+    date = metadata_harvester_cli.last_harvest_date(create_test_log_file)
+    result = metadata_harvester_cli.records_to_dict(date)
+    assert date == "2023-09-08T14:45:58Z"
+    assert single_record_to_dict == result
+
+
+def test_records_to_dict_without_last_harvest_date(single_record_to_dict, create_test_log_file_with_unsuccessful_harvest):
+    """
+    Test that fetching records without a log file succeeds (all records are fetched).
+    """
+    date = metadata_harvester_cli.last_harvest_date(create_test_log_file_with_unsuccessful_harvest)
+    result = metadata_harvester_cli.records_to_dict(date)
+    assert date is None
+    assert single_record_to_dict == result
+
+
+@pytest.fixture
+def create_test_log_file():
+    """Create a temporary log file for testing and clean up afterwards."""
+    log_file = "harvester_test.log"
+    log_file_data = [
+        "2023-09-08 14:34:16,887 - INFO - Started\n"
+        "2023-09-08 14:42:58,652 - INFO - Success, all records harvested\n"
+        "2023-09-08 14:44:58,690 - INFO - Started\n"
+        "2023-09-08 14:45:58,690 - INFO - Started\n"
+        "2023-09-08 14:45:58,956 - INFO - Success, records harvested since 2023-09-08T14:34:16Z\n"
+    ]
+    with open(log_file, "w") as file:
+        file.writelines(log_file_data)
+    yield log_file
+    os.remove(log_file)
```

Comment:
We don't need to take care of manual file creation and deletion in the current directory (which can cause problems like overwriting content or the test file not getting removed if the tests crash) if we use the pre-made [tmp_path](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html) fixture. With that we could do something like
```
log_file = tmp_path / "harvester_test.log"
log_file.writelines(log_file_data)
return log_file
```

The same goes for other places where we manually create files.